src/main/scala/controllers/ActionHandler.scala

```diff
@@ -102,27 +102,22 @@ class ActionHandler @Inject()(applications: ApplicationOps, applicationForms: Ap
   def redirectToPreview(id: ApplicationId, sectionNumber: Int) =
     Redirect(routes.ApplicationPreviewController.previewSection(id, sectionNumber))
 
-  def renderSectionForm(id: ApplicationId,
+  def renderSectionForm(app: ApplicationDetail,
```

Comment:
But that's not a problem. If something can supply the right parameters to call the function without making a call that returns a Future (say, a unit test) then why shouldn't it just return a Result? Controllers are free to return non-Future-wrapped results if it's appropriate.